sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -117,6 +117,8 @@ public class SystemSchema extends AbstractSchema
           segment.getDataSource())
       );
 
+  private static final long REPLICATION_FACTOR_UNKNOWN = -1L;
```

Comment:
From what I could find, integer columns do not allow null. I change it to a string column, which does allow it, but this would return integers as strings as well, if that is preferable.

I think the console currently does a little work in beautifying the details, so we could automatically show -1 as "No details available" ( I might be wrong, but I think it already does this for the shard spec column)

Reply:
Yeah, makes sense, let's keep it as -1 if no information is available and 0 for overshadowed segments.